thirdparty/libart/src/art.c

```diff
@@ -21,6 +33,44 @@ static uint32_t __inline __builtin_ctz(uint32_t x) {
 }
 #endif
 
+// The ART implementation requires that no key be a full prefix of an existing
+// tree during insertion.  In practice this means that each key must have a
+// terminator character.  One approach is to ensure that the key and key_len
+// includes a physical trailing NUL terminator when inserting C-strings.
+// This doesn't help a great deal when working with binary strings that may be
+// a slice in the middle of a buffer that has no termination.
```

Comment:
Yeah, not so much at insertion time, but when we perform a prefix search we don't want to have to make a copy, especially if there are no matches for the prefix.